python/arcticdb/version_store/library.py

```diff
@@ -1706,6 +1706,20 @@ def reload_symbol_list(self):
         """
         self._nvs.version_store.reload_symbol_list()
 
+    def compact_symbol_list(self) -> None:
+        """
+        Compact the symbol list cache into a single key in the storage
+
+
+        Raises
+        ------
+        PermissionException
+            Library has been opened in read-only mode
+        InternalException
+            Storage lock required to compact the symbol list could not be acquired
+        """
+        self._nvs.compact_symbol_list()
```

Comment:
Can it return something to indicate whether it did anything to storage, or how much compaction it did?

Reply:
The method will always compact the symbol list cache down to a single key or throw. I've updated to return the number of symbol list keys prior to the compaction though